tests/unit/model/session_test.py

```diff
@@ -2629,17 +2632,444 @@ def test_state_normalization(
+                            is_definite=True,
+                        ),
+                    )
+                ],
+                actions=[stmt.Append("Msg", expr.Variable("E"))],
+                transitions=[Transition(target=ID("End"))],
+            ),
+            State(
+                "S",
+                declarations=[
+                    decl.VariableDeclaration(
+                        "Msg",
+                        "Message",
+                        type_=rty.Message(
+                            "M",
+                            is_definite=True,
+                        ),
+                    )
+                ],
+                actions=[stmt.Append("Msg", expr.Variable("E"))],
+                transitions=[Transition(target=ID("End"))],
+            ),
+        ),
+        (
+            State(
+                "S",
+                declarations=[
+                    decl.VariableDeclaration(
+                        "Msg",
+                        "Message",
+                        type_=rty.Message(
+                            "M",
+                            is_definite=True,
+                        ),
+                    )
+                ],
+                actions=[stmt.Append("List", expr.Variable("Msg"))],
+                transitions=[Transition(target=ID("End"))],
+            ),
+            State(
+                "S",
+                declarations=[
+                    decl.VariableDeclaration(
+                        "Msg",
+                        "Message",
+                        type_=rty.Message(
+                            "M",
+                            is_definite=True,
+                        ),
+                    )
+                ],
+                actions=[stmt.Append("List", expr.Variable("Msg"))],
+                transitions=[Transition(target=ID("End"))],
+            ),
+        ),
+        (
+            State(
+                "S",
+                declarations=[
+                    decl.VariableDeclaration(
+                        "Msg",
+                        "Message",
+                        type_=rty.Message(
+                            "M",
+                            is_definite=True,
+                        ),
+                    )
+                ],
+                actions=[stmt.Extend("List", expr.Variable("Msg"))],
+                transitions=[Transition(target=ID("End"))],
+            ),
+            State(
+                "S",
+                declarations=[
+                    decl.VariableDeclaration(
+                        "Msg",
+                        "Message",
+                        type_=rty.Message(
+                            "M",
+                            is_definite=True,
+                        ),
+                    )
+                ],
+                actions=[stmt.Extend("List", expr.Variable("Msg"))],
+                transitions=[Transition(target=ID("End"))],
+            ),
+        ),
```

Comment:
Another test case which seems invalid to me. A sequence cannot be extended by a message.

There are similar problems in the subsequent test cases. I think it would be better to not use examples which would be rejected during type checking.

Reply:
Removed.